python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -498,52 +514,90 @@ def explain_node(
+            nodes = nodes.tolist()
+        if isinstance(nodes, int):
+            nodes = [nodes]
+
         self.model = self.model.to(graph.device)
         self.elayers = self.elayers.to(graph.device)
 
-        embed = self.model(graph, feat, embed=True, **kwargs)
-        embed = embed.data
+        batched_graph = []
+        batched_feats = []
+        batched_embed = []
+        batched_node_ids = []
+        node_idx = 0
+        for node_id in nodes:
+            sg, inverse_indicies = khop_in_subgraph(
+                graph, node_id, self.num_hops
+            )
+            sg_feat = feat[sg.ndata[NID].long()]
 
-        edge_idx = graph.edges()
+            embed = self.model(sg, sg_feat, embed=True, **kwargs)
+            embed = embed.data
 
-        col, row = edge_idx
-        col_emb = embed[col.long()]
-        row_emb = embed[row.long()]
-        self_emb = embed[node_id].repeat(graph.num_edges(), 1)
-        emb = torch.cat([col_emb, row_emb, self_emb], dim=-1)
+            col, row = sg.edges()
+            col_emb = embed[col.long()]
+            row_emb = embed[row.long()]
+            self_emb = embed[inverse_indicies[0]].repeat(sg.num_edges(), 1)
+            emb = torch.cat([col_emb, row_emb, self_emb], dim=-1)
+            batched_embed.append(emb)
+            batched_graph.append(sg)
+            batched_feats.append(sg_feat)
+            batched_node_ids.append(inverse_indicies[0] + node_idx)
+            node_idx += sg.num_nodes()
 
-        emb = self.elayers(emb)
-        values = emb.reshape(-1)
+        batched_graph = batch(batched_graph)
+        batched_feats = torch.cat(batched_feats)
+        batched_embed = torch.cat(batched_embed)
+
+        batched_embed = self.elayers(batched_embed)
+        values = batched_embed.reshape(-1)
 
         values = self.concrete_sample(values, beta=tmp, training=training)
         self.sparse_mask_values = values
 
-        reverse_eids = graph.edge_ids(row, col).long()
+        col, row = batched_graph.edges()
+        reverse_eids = batched_graph.edge_ids(row, col).long()
         edge_mask = (values + values[reverse_eids]) / 2
 
-        self.set_masks(graph, edge_mask)
+        self.set_masks(batched_graph, edge_mask)
 
         # the model prediction with the updated edge mask
-        logits = self.model(graph, feat, edge_weight=self.edge_mask, **kwargs)
+        logits = self.model(
+            batched_graph, batched_feats, edge_weight=self.edge_mask, **kwargs
+        )
         probs = F.softmax(logits, dim=-1)
 
         if not training:
             self.clear_masks()
 
-        return (probs, edge_mask) if training else (probs.data, edge_mask)
+        return (
+            (probs, edge_mask, batched_graph, batched_feats, inverse_indicies)
+            if training
+            else (
+                probs.data,
+                edge_mask,
+                batched_graph,
+                batched_feats,
+                inverse_indicies,
```

Comment:
It doesn't look correct to me to return `inverse_indicies` for the last node in the list.

Reply:
sorry, it should be the collection of `inverse_indices` from each khop subgraph adjusted for the batching